application/F3DStarter.cxx

```diff
@@ -322,6 +323,27 @@ int F3DStarter::Start(int argc, char** argv)
     }
 #endif
   }
+
+  // Parse colormap
+  if (!this->Internals->AppOptions.ColorMapFile.empty())
+  {
+    std::string fullPath = F3DColorMapTools::Find(this->Internals->AppOptions.ColorMapFile);
+
+    if (!fullPath.empty())
+    {
+      auto cm = F3DColorMapTools::Read(fullPath);
+
+      if (!cm.empty())
+      {
+        this->Internals->Engine->getOptions().set("model.scivis.colormap", cm);
```

Comment:
Something I dont like about this is that `colormap-file` only takes precedence if it is valid. You should add actual codepath to handle "both defined" case imo.

Reply:
The problem is there is a default value for the colormap so I cannot rely on the f3d::options value. And I don't have access to the cxxopts args here.

Reply:
how about cleaning up `colormap` if `colormap-file` is not empty ?